File: android_fragment/app/src/main/java/com/support/generator/module/main/MainActivityPresenter.java
package com.support.generator.module.main;

import com.support.generator.common.Presenter;

import javax.inject.Inject;

/**
 * @author: leellun
 * @data: 2018/12/24.
 */
public class MainActivityPresenter
        extends Presenter<MainActivityContract.IMainView> implements MainActivityContract.IMainPresenter {
  @Inject
  public MainActivityPresenter() {
  }

  public void onViewAttached(MainActivityContract.IMainView iview, boolean created) {
    super.onViewAttached(iview, created);
  }

  public void onViewDetached() {
    super.onViewDetached();
  }

  public void onDestroy() {
    super.onDestroy();
  }
}

File: example/app/src/main/java/gauges/mobile/github/com/mvpapplication/module/main/MainActivityPresenter.kt
package gauges.mobile.github.com.mvpapplication.module.main

import gauges.mobile.github.com.mvpapplication.common.mvphelper.Presenter
import javax.inject.Inject

/**
 * @author: leellun
 * @data: 2018/12/19.
 *
 */
class MainActivityPresenter @Inject constructor()
    : Presenter<MainActivityContract.IMainView>(), MainActivityContract.IMainPresenter {

    override fun onViewAttached(iview: MainActivityContract.IMainView, created: Boolean) {
        super.onViewAttached(iview, created)
    }

    override fun onViewDetached() {
        super.onViewDetached()
    }

    override fun onDestroy() {
        super.onDestroy()
    }
}